### DISCUSSION TITLE
Does the query rows method in automations have a hard limit of 1000 rows?

### DISCUSSION THREAD

**Discussion by FlaminWrap** [Budibase/budibase#5685] [Help] [answered]
Hi Folks,

Does the Query Rows step in automations have a limit of 1000 rows? I am trying to do a count of all the applications for the month of April. March worked fine but April is maxing out at 1000 (there has been over 1000 applications in April). Please see the screenshot below for the results and setup.

Result
![image](https://user-images.githubusercontent.com/97764630/166441327-62327845-3455-47ec-9534-bb36d3e057ec.png)

Query rows setup
![image](https://user-images.githubusercontent.com/97764630/166441110-2d4dad48-0d42-419d-946b-c8ef7031a54a.png)

Thanks.

**Comment by melohagan** [comment, marked as answer]
Hey @FlaminWrap 

Are you hosted in the cloud? 
There is a limit of 1000 automation operations: https://budibase.com/pricing/

**Reply by FlaminWrap** [reply]
Self hosted

**Reply by melohagan** [reply]
Yep there's a limit of 1000:
![Screenshot 2022-05-03 at 14 00 38](https://user-images.githubusercontent.com/101575380/166457473-ca2d969c-a53b-42d5-b2da-d56cef259451.png)

That's for the internal CouchDB table. There shouldn't be a limit for external tables, e.g. SQL

**Reply by FlaminWrap** [reply]
I see. I will keep this in mind for future projects and look to export the data to mysql or something.

**Reply by mike12345567** [reply]
Hi @FlaminWrap - this limit exists for all searching within the internal DB, anything more than 1000 rows returned at once can be problematic for our Lucene index. You raise a good point however that for automations you can't easily paginate, meaning that it is difficult to work on large amounts of data stored in the internal DB - this is something we will need to have a think about.

**Reply by FlaminWrap** [reply]
Thanks for the clarification @mike12345567.
